### PR TITLE
Improvements to the default screen experience

### PR DESCRIPTION
**Pull request by mehalshah**
-Took out the data-is-default tag, was making things needlessly complex and wasn't backwards compatible with older apps
-Button doesn't show up if you are on the default screen
-Changed tests

### CODE REVIEW

File: apps/src/applab/designElements/DefaultScreenButtonPropertyRow.jsx [hunk 1]
```diff
@@ -6,6 +7,11 @@ var DefaultButtonPropertyRow = React.createClass({
   },
 
   render: function () {
+    if (elementUtils.getId(document.querySelector('#designModeViz .screen')) ===
+      this.props.screenId) {
```
Comment (Bjvanminnen) [inline apps/src/applab/designElements/DefaultScreenButtonPropertyRow.jsx]: nit: style is to indent this line an extra two chars to differentiate from contents within the block

**Comment by Bjvanminnen** [comment]
Do we have to worry about projects created in the interim, where data-is-default got set?

**Comment by mehalshah** [comment]
They'll have an attribute created that does nothing. If we want to bulk edit the source, we can do that, but I don't think its that important.

File: apps/src/applab/designElements/DefaultScreenButtonPropertyRow.jsx (hunk 1 above)
Comment (mehalshah) [inline apps/src/applab/designElements/DefaultScreenButtonPropertyRow.jsx]: Will fix

**Comment by Bjvanminnen** [comment]
Is the default screen in that case guaranteed to be first in the DOM order? I can't remember if we also moved it or not.

**Comment by mehalshah** [comment]
The first screen is treated as the default - yes.

**Comment by Bjvanminnen** [comment]
lgtm

**Pull request merged**